pkg/server/container_create_windows.go

```diff
@@ -70,11 +70,17 @@ func (c *criService) containerSpec(id string, sandboxID string, sandboxPid uint3
 
 	specOpts = append(specOpts, customopts.WithWindowsResources(config.GetWindows().GetResources()))
 
-	username := config.GetWindows().GetSecurityContext().GetRunAsUsername()
-	if username != "" {
-		specOpts = append(specOpts, oci.WithUser(username))
+	securityContext := config.GetWindows().GetSecurityContext()
+	if securityContext != nil {
+		username := securityContext.GetRunAsUsername()
```

Comment:
```
// WindowsContainerSecurityContext holds windows security configuration that will be applied to a container.
type WindowsContainerSecurityContext struct {
	// User name to run the container process as. If specified, the user MUST
	// exist in the container image and be resolved there by the runtime;
	// otherwise, the runtime MUST return error.
	RunAsUsername string `protobuf:"bytes,1,opt,name=run_as_username,json=runAsUsername,proto3" json:"run_as_username,omitempty"`
	// The contents of the GMSA credential spec to use to run this container.
```
I read that as: 
```
if username != "" && username != imageConfig.User { return nil, errors.Wrap(err, "failed to match username with image username")}
```

Reply:
if so.. we should probably add that code ^ and test over here:
https://github.com/containerd/cri/blob/master/pkg/server/container_create_windows_test.go

to make sure all is well with the CRI rule above.

Reply:
@mikebrow Sorry for not getting back to this yesterday. So there's a couple things regarding this. 

1. We can't just check if the user name isn't equal to the imageConfigs user as that would inhibit running as a local system account/another default container user (ContainerUser and ContainerAdministrator are the two default virtual accounts) if this wasn't what's present in the imageConfig. For example, on nanoserver:1809 the image config user is specified as ContainerUser, but ContainerAdministrator and the local accounts (NT AUTHORITY\NETWORK SERVICE etc) are all valid options as well. On our end in hcsshim we actually do error out when trying to create the init process with a user that doesn't exist in the container so this case is handled, just later down in the stack. 

2. What can be changed however is that it looks like we don't currently uphold the image config user if no override is passed, it seems we just default to ContainerAdministrator. Going through cri and running nanoserver, the container user ends up being ContainerAdministrator when it truthfully should be ContainerUser if we want parity with dockers behavior. So we can check if there is a user in the image config --> if not then leave as is and default to ContainerAdministrator, otherwise set this as spec.Process.User.Username unless RunAsUsername is specified. I can then add a test for this behavior. 

Let me know if this sounds good!